### PR TITLE
Add matplotlib plotting to VertexClustering

### LINKED ISSUE #370: Plotting VertexClustering fails when targeting matplotlib
**Describe the bug**
Ordinarily, one could plot a `VertexClustering`, for example using

```python
G = ig.Graph.Famous('Zachary')
comms = G.community_leiden('modularity')
ig.plot(comms)
```
image

However, when doing `ig.plot(comms, target=ax)`, where `ax` is a `matplotlib` `Axes`, this throws the error
`AttributeError: 'VertexClustering' object has no attribute 'layout'`.

**To reproduce**
```python
import igraph as ig
import matplotlib.pyplot as plt
G = ig.Graph.Famous('Zachary')
comms = G.community_leiden('modularity')
fig, ax = plt.subplots()
ig.plot(comms, target=ax)
```

**Version information**
`python-igraph` version 0.9.0, linked against external shared `igraph` library version 0.9.0-135-g4019b1f29.

### PR DESCRIPTION
**Pull request by iosonofabio**
This PR aims to fix #370 

ATM there is no specific test for this.

### CODE REVIEW

**Comment by ntamas** [comment]
It's hard to test anyway, so let's just roll with the implementation. Let me know when it's ready for review.

**Comment by iosonofabio** [comment]
The whole "mark_groups" option is a little weird when used in conjunction with a `VertexClustering` as the first argument of `igraph.plot`, but changing that breaks API for the Cairo backend, so let's not touch it here.

Ready for merge.

**Comment by vtraag** [comment]
> The whole "mark_groups" option is a little weird when used in conjunction with a `VertexClustering` as the first argument of `igraph.plot`, but changing that breaks API for the Cairo backend, so let's not touch it here.

Well, yes, sort of. A simple boolean would suffice, simply to indicate whether you want to draw the convex hulls or not.

**Comment by iosonofabio** [comment]
Fair enough, I didn't know that was an option since it's not mentioned in the docs. I added it and also cleaned up the code a little, and added this info to the docs.

I then rebased on top of the current master, so if it passes CI it should be good to go.

File: src/igraph/drawing/graph.py [hunk 1]
```diff
@@ -1010,6 +1012,61 @@ def callback_edge_offset(event):
...
+
+        # mark groups: the used data structure is eventually the dict option:
+        # tuples of vertex indices as the keys, colors as the values. We
+        # convert other formats into that one here
+        if "mark_groups" in kwds:
+            if kwds["mark_groups"] is False:
+                del kwds["mark_groups"]
+            elif (kwds["mark_groups"] is True) and (clustering is not None):
+                pass
+            # Deferred import to avoid a cycle in the import graph
```
Comment (ntamas) [inline src/igraph/drawing/graph.py:1031]: The deferred import is not here but above

File: src/igraph/drawing/graph.py [hunk 2]
```diff
@@ -1010,6 +1012,61 @@ def callback_edge_offset(event):
...
+                    membership = [x[0] for x in membership]
+                clusters = sorted(set(membership))
+                n_colors = len(clusters)
+                cmap = mpl.cm.get_cmap("viridis")
+                colors = [cmap(1.0 * i / n_colors) for i in range(n_colors)]
+                c = [colors[clusters.index(i)] for i in membership]
+                kwds["vertex_color"] = c
+
+                # mark_groups if not explicitely marked
+                if ("mark_groups" in kwds) and (kwds["mark_groups"] is True):
```
Comment (ntamas) [inline src/igraph/drawing/graph.py:1061]: `if kwds.get("mark_groups") is True` is enough; `.get()` returns `None` if the key is not in the dict

**Review by ntamas — APPROVED** [review]
Okay, seems good, only a few minor changes were requested but I'll do these myself.

**Pull request merged**